devito/equation.py

```diff
@@ -87,9 +96,23 @@ def evaluate(self):
             except AttributeError:
                 if bool(rules):
                     eq = eq.xreplace(rules)
-
         return eq
 
+    @property
+    def _flatten(self):
+        """
+        Flatten vectorial/tensorial Equation into list of scalar Equations.
+        """
+        try:
```

Comment:
I don't like try-except when we can get away with a clear `if is_Matrix ... `

also, the except is missing the exception